### PR TITLE
add node10 to core-dev images

### PR DESCRIPTION
**Pull request by benoitvidis**
## What does this PR do ?

Add node.js 10.16.0 to core-dev* images.

### Other changes

Merge the `RUN` steps between node install and the rest to allow apt cache deletion.

### CODE REVIEW

File: core-dev/Dockerfile [hunk 1]
```diff
@@ -7,39 +7,41 @@ WORKDIR /var/app
...
       gdb \
       git \
+      gnupg \
       libfontconfig \
       libzmq3-dev \
       python \
       procps \
       wget \
-      nodejs \
+  && curl -sL https://deb.nodesource.com/setup_8.x | bash - \
```
Comment (Aschen) [inline core-dev/Dockerfile]: I think we could already use node 10 to download `n` and install other versions

File: core-dev/Dockerfile [hunk 2]
```diff
@@ -7,39 +7,41 @@ WORKDIR /var/app
...
-    pm2 \
   && apt-get clean \
   && rm -rf /var/lib/apt/lists/* \
   && echo "alias ll=\"ls -lahF --color\"" >> ~/.bashrc \
-  && chmod +x ~/.bashrc
-
-RUN n $NODE_8_VERSION
+  && chmod +x ~/.bashrc \
+  \
+  && n $NODE_8_VERSION
```
Comment (Aschen) [inline core-dev/Dockerfile]: Why are you removing a layer ? We are gonna need to rebuild almost the entire image to change node default version

Comment (benoitvidis) [reply core-dev/Dockerfile]: Yes, but is it a real problem? 
I prefer to stick to the general Docker guideline of [minimizing the number of layers](https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#decouple-applications).

Comment (Aschen) [reply core-dev/Dockerfile]: Oh ok I didn't know that this was the official guidelines

**Pull request merged**